### ISSUE TITLE
Add `multiclass` fine-tuning example

### ISSUE THREAD

**Issue by qubvel** [qubvel-org/segmentation_models.pytorch#911]
It would be great to have more examples for fine-tuning in the library!

Similar to current examples for [binary-segmentaion](https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/examples/binary_segmentation_intro.ipynb) and [multi-label](https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/examples/cars%20segmentation%20(camvid).ipynb) segmentation would be great to have multi-class segmentation example (the whole image is annotated). The same camvid dataset can be used.

The example should showcase how to 
 - fine-tune a model with pytorch-ligtning (or any other fine-tuning framework, even a plain pytorch)
 - use loss function with ignore index for unlabeled class
 - compute metrics per-class
 - visualize results

In case anyone wants to work on this you are welcome! Just notify in this issue, and then ping me fir review a PR when it's ready.

**Comment by abdksyed** [comment]
I can take this up.

Do we want to fine-tune on all the classes or should I limit to say 5 most common classes from the dataset?

**Comment by qubvel** [comment]
Hi @abdksyed, thanks a lot for your interest to this issue!
Let's have all the classes from the dataset.

**Comment by TimbusCalin** [comment]
@qubvel I have managed to create a multiclass classsification example. 

I am have created a local branch and when I try to push to origin, I get fatal: unable to access 'https://github.com/qubvel-org/segmentation_models.pytorch.git/': The requested URL returned error: 403.

Can you set me as a collaborator in order that I can open a PR?

**Comment by abdksyed** [comment]
looks like I am little late. I was about to create a PR, with the notebook. But looks like @TimbusCalin has already done it.

**Comment by qubvel** [comment]
Hi @abdksyed, If you already have the work done you can open a PR, I will prioritize it because you were first on this issue (sorry @TimbusCalin). Or, to save your contribution, you can implement additional features, like per-clas metrics computing for multiclass segmentation, and join with @TimbusCalin PR.

Next time I would recommend opening a PR in draft mode and linking it to the issue as early as possible to let others know that work is in progress 🙂 

Thank you both for your contributions, I hope we will resolve this issue in a good way for both of you!

**Comment by abdksyed** [comment]
I am happy with @TimbusCalin's PR, we can merge that. As for the class wise metric, I will try to implement it, but if @TimbusCalin does it before me, he can push the update.

**Comment by TimbusCalin** [comment]
Yes @qubvel we also had a brief discussion in the PR thread where we decided that it's optimum to integrate the PR now, and then @abdksyed  will take on to implement the class-wise metric.
